### PR TITLE
Add the feature of showing useful error message during CI

### PR DESCRIPTION
**Pull request by RyanL1997**
Signed-off-by: Ryan Liang <jiallian@amazon.com>

### Description
Create a separate workflow and implemented environment variable of version to check for the existence of the OpenSearch Security Plugin artifact

### Issues Resolved
* Resolved https://github.com/opensearch-project/security-dashboards-plugin/issues/1073 

### Testing
~[CI test case#1 with "403 Forbidden" response](https://github.com/opensearch-project/security-dashboards-plugin/runs/8000467600?check_suite_focus=true)~
~[CI test case#2 with "404 Not Found" response](https://github.com/opensearch-project/security-dashboards-plugin/runs/8000584940?check_suite_focus=true)~
Created a new workflow for "Verify there is matching build of security plugin" by running `prerequisites-test`

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by cliu123 — APPROVED** [review]

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -16,6 +16,9 @@ jobs:
           wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.3.0/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-2.3.0-linux-x64.tar.gz
           tar -xzf opensearch-*.tar.gz
           rm -f opensearch-*.tar.gz
+
+      - name: Check the validation of OpenSearch Security Plugin
+        run: wget -S --spider https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.3.0/latest/linux/x64/tar/builds/opensearch/plugins/opensearch-security-2.3.0.0.zip || (echo "Please make sure security plugin has been bumped to the same version and added to manifest." && exit 1)
```
Comment (cwperks) [inline .github/workflows/integration-test.yml:21]: Is it possible to create variables in the Github actions files? We use the version in a number of places and it would be nice to define it once.

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -16,6 +16,9 @@ jobs:
           wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.3.0/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-2.3.0-linux-x64.tar.gz
           tar -xzf opensearch-*.tar.gz
           rm -f opensearch-*.tar.gz
+
+      - name: Check the validation of OpenSearch Security Plugin
```
Comment (peternied) [inline .github/workflows/integration-test.yml:20]: This would fail the `integration-test` workflow, what would you think about creating a separate workflow called something like `prerequisites` that could have a task named "Verify there is matching build of security plugin"

File: .github/workflows/integration-test.yml (hunk 1 above)
Comment (RyanL1997) [reply .github/workflows/integration-test.yml:21]: Thanks for the reply. I think we can only set up the environment variables in the Github actions files. I can definitely try to set it up in the environment variable to see if it works.

File: .github/workflows/integration-test.yml (hunk 2 above)
Comment (RyanL1997) [reply .github/workflows/integration-test.yml:20]: Thanks for the reply! The reason I set to check it up here because if this one fails the next test of `Download OpenSearch Security Plugin` will also be failed and break the `integration-test`. I was planning to use this test only as the prerequisite for checking the test of `Download OpenSearch Security Plugin`.  Sorry for the confusions anyways. Please correct me if I miss some of the steps.

Comment (peternied) [reply .github/workflows/integration-test.yml:20]: Here is the UI that our users would see on failures, might be useful to have this spelled out more plainly (Less clicks is better!)
image

Comment (RyanL1997) [reply .github/workflows/integration-test.yml:20]: I think I get it now! Ya I will create a new work flow with a better name for running this separately!  Thanks @peternied!

**Comment by RyanL1997** [comment]
He guys, I have made the update of this PR:
  1. Instead of directly to implement the test in `integration-test`, I created a separate workflow called `prerequisite-test`.
  2. Implemented the environment variable of version numbers in `prerequisite-test`
Seems like it is also possible to setup the environment variable of version in the `Integration-test`. However, we may need another PR for doing that if it is necessary.

File: .github/workflows/integration-test.yml [hunk 3]
```diff
@@ -16,7 +16,7 @@ jobs:
           wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.3.0/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-2.3.0-linux-x64.tar.gz
           tar -xzf opensearch-*.tar.gz
           rm -f opensearch-*.tar.gz
-          
+
```
Comment (cliu123) [inline .github/workflows/integration-test.yml:19]: Please remove this change.

**Comment by cliu123** [comment]
@RyanL1997 Would you please add the new behavior in the PR description?

File: .github/workflows/integration-test.yml [hunk 4]
```diff
@@ -16,7 +16,6 @@ jobs:
           wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/2.3.0/latest/linux/x64/tar/builds/opensearch/dist/opensearch-min-2.3.0-linux-x64.tar.gz
           tar -xzf opensearch-*.tar.gz
           rm -f opensearch-*.tar.gz
-
```
Comment (cliu123) [inline .github/workflows/integration-test.yml:19]: Please remove changes in this file. This file is not supposed to be present in `Files changes`.

File: .github/workflows/integration-test.yml (hunk 3 above)
Comment (RyanL1997) [reply .github/workflows/integration-test.yml]: Ya I have reset it to main and pushed again. Thx for the review @cliu123!

File: .github/workflows/integration-test.yml (hunk 4 above)
Comment (RyanL1997) [reply .github/workflows/integration-test.yml]: Ya I have reset it to main and pushed again. Thx for the review @cliu123!

File: .github/workflows/prerequisite-test.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+name: Verify there is matching build of security plugin
+
+on: [push, pull_request]
+
+jobs:
+  tests:
+    name: Run prerequisite tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check the validation of OpenSearch Security Plugin
```
Comment (cwperks) [inline .github/workflows/prerequisite-test.yml:10]: The name of this should reflect what the check is performing. How does "Check for the existence of the OpenSearch Security Plugin artifact" sound?

Comment (RyanL1997) [reply .github/workflows/prerequisite-test.yml]: Sure, I have updated the commit and also changed the PR description with that. Thx for the review @cwperks!

File: .github/workflows/prerequisite-test.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+name: Verify there is matching build of security plugin
```
Comment (cwperks) [inline .github/workflows/prerequisite-test.yml:1]: This name appears in the checks like this:

Screen Shot 2022-08-24 at 4 44 09 PM

Shorten this to "Prerequisite Checks"?

File: .github/workflows/prerequisite-test.yml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  tests:
+    name: Run prerequisite tests
```
Comment (cwperks) [inline .github/workflows/prerequisite-test.yml:7]: Similar to above: maybe "Run prerequisite _checks_"?

Comment (RyanL1997) [reply .github/workflows/prerequisite-test.yml]: Thanks! I have changed the name for both test's name and the sub tests' name. Please review it.

**Review by cliu123 — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

**Review by cliu123 — APPROVED** [review]
